### PR TITLE
Implement Virtfs Thin interface

### PR DESCRIPTION
**Pull request by movitto**

### CODE REVIEW

File: lib/virtfs/iso9660/base.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
+require 'rufus-lru'
+require 'virtfs/nativefs/thin'
```
Comment (roliveri) [inline lib/virtfs/iso9660/base.rb:2]: Why is this being required?

File: lib/virtfs/iso9660/base.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    def drive_root
+      @drive_root ||= Directory.new(boot_sector, drive_root_entry)
+    end
+
+    # Wack leading drive leter & colon.
+    def unnormalize_path(p)
+      p[1] == 58 ? p[2, p.size] : p
+    end
+  end # class FS
```
Comment (roliveri) [inline lib/virtfs/iso9660/base.rb:80]: Is the `thin_interface?` method defined anywhere?

File: lib/virtfs/iso9660/base.rb (hunk 1 above)
Comment (movitto) [inline lib/virtfs/iso9660/base.rb:2]: To pull in the thin interface, IIUC it's one of the two virtfs interfaces which each module must separately pull in & implement the details.

If this is not required let me know and  I can remove it from the PR & rebase the others

File: lib/virtfs/iso9660/base.rb (hunk 2 above)
Comment (movitto) [inline lib/virtfs/iso9660/base.rb:80]: @roliveri yes but in the last / cleanup PR which flushes out the odds and ends to make things work end to end

https://github.com/ManageIQ/virtfs-iso9660/pull/10/commits/49283f0490e08dbb028c1b247f2c9a992f424576#diff-2b0d12f2649db8eb256e9aedaa454b64R13

File: lib/virtfs/iso9660/base.rb (hunk 1 above)
Comment (roliveri) [inline lib/virtfs/iso9660/base.rb:2]: No, there's no need to require that. That's the implementation of the thin interface to the native file system, not an interface definition that you need to pull in. You just need to implement the thin interface methods in your classes.

**Comment by movitto** [comment]
@roliveri closing this PR & replacing with #11 which pulls this in and the consolidated core-implementation logic from the subsequent PRs. Also rebased those so as to just include their target functionality

**Pull request closed without merging**